src/controllers/confirm.company.controller.ts

```diff
@@ -21,23 +22,29 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = async (req: Request, res: Response) => {
+export const post = async (req: Request, res: Response, next: NextFunction) => {
   const session: Session = req.session as Session;
   const companyNumber = req.query["companyNumber"] as string;
-  const eligibilityStatusCode: EligibilityStatusCode = await checkEligibility(session, companyNumber);
-  if (eligibilityStatusCode === EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE) {
-    await createNewConfirmationStatement(req);
-    const url = TRADING_STATUS_PATH.replace(":companyNumber", companyNumber);
-    return res.redirect(url);
-  } else {
-    return displayEligibilityStopPage(res, eligibilityStatusCode );
+  try {
+    const eligibilityStatusCode: EligibilityStatusCode = await checkEligibility(session, companyNumber);
+    if (eligibilityStatusCode === EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE) {
+      await createNewConfirmationStatement(req);
+      const url = TRADING_STATUS_PATH.replace(":companyNumber", companyNumber);
+      return res.redirect(url);
+    } else {
+      return displayEligibilityStopPage(res, eligibilityStatusCode);
+    }
+  } catch (e) {
+    logger.errorRequest(req, `An error has occurred. Re-routing to the error screen - ${e.stack}`);
+    return next(e);
   }
 };
 
 const displayEligibilityStopPage = (res: Response, eligibilityStatusCode: EligibilityStatusCode) => {
   if (eligibilityStatusCode === EligibilityStatusCode.INVALID_COMPANY_STATUS) {
     return res.render(Templates.INVALID_COMPANY_STATUS);
   }
+  throw new Error("Unknown eligibility code");
```

Comment:
can you include the eligibilityStatusCode in the message so when it happens in live we will know what caused the failure.